staging/src/k8s.io/client-go/dynamic/interface.go

```diff
@@ -57,3 +58,19 @@ func LegacyAPIPathResolverFunc(kind schema.GroupVersionKind) string {
 	}
 	return "/apis"
 }
+
+// Lister helps list resources.
+type Lister interface {
+	// List lists all resources in the indexer.
+	List(selector labels.Selector) (ret []*unstructured.Unstructured, err error)
```

Comment:
not sure if we want to provide a way to list all items or only the ones from a namespace.

Reply:
> not sure if we want to provide a way to list all items or only the ones from a namespace.

I think we have to provide both in this lister since we don't know which is applicable.  We probably need a top level Get too.